internal/v1/handler_blueprints.go

```diff
@@ -308,10 +312,54 @@ func (h *Handlers) ExportBlueprint(ctx echo.Context, id openapi_types.UUID) erro
 		Distribution:   blueprint.Distribution,
 		Customizations: blueprint.Customizations,
 		Metadata: BlueprintMetadata{
-			ExportedAt: time.Now().UTC().String(),
+			ExportedAt: now().UTC().String(),
 			ParentId:   &id,
 		},
 	}
+
+	repoUUIDs := []string{}
+	if blueprint.Customizations.CustomRepositories != nil {
+		for _, repo := range *blueprint.Customizations.CustomRepositories {
+			repoUUIDs = append(repoUUIDs, repo.Id)
+		}
+	}
+
+	exportedRepositoriesResp, err := h.server.csClient.BulkExportRepositories(ctx.Request().Context(), content_sources.ApiRepositoryExportRequest{
+		RepositoryUuids: common.ToPtr(repoUUIDs),
+	})
+	if err != nil {
+		return err
+	}
+	defer closeBody(ctx, exportedRepositoriesResp.Body)
+
+	if exportedRepositoriesResp.StatusCode != http.StatusOK {
+		if exportedRepositoriesResp.StatusCode != http.StatusUnauthorized {
+			body, err := io.ReadAll(exportedRepositoriesResp.Body)
+			if err != nil {
+				return err
+			}
+			ctx.Logger().Warnf("Unable to export custom repositories: %s", body)
+		}
+		return fmt.Errorf("Unable to fetch custom repositories, got %v response", exportedRepositoriesResp.StatusCode)
+	}
+
+	if exportedRepositoriesResp.Body != nil {
+		bodyBytes, err := io.ReadAll(exportedRepositoriesResp.Body)
+		if err != nil {
+			return fmt.Errorf("Unable to export custom repositories: %w", err)
+		}
+
+		if len(bodyBytes) != 0 {
+			// Saving the custom repositories in the object format
+			var result []map[string]interface{}
+			err = json.Unmarshal(bodyBytes, &result)
+			if err != nil {
+				return fmt.Errorf("Unable to export custom repositories: %w, %s", err, string(bodyBytes))
+			}
+			blueprintExportResponse.ContentSources = &result
+		}
+	}
```

Comment:
I suggest to concatenate all three if statement into one, then the read all code will not be duplicated and there will be only one level of identation.